src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php

```diff
@@ -165,9 +165,13 @@ private function registerHandlers(ContainerBuilder $container, array $busIds): v
         $container->addDefinitions($definitions);
 
         foreach ($busIds as $bus) {
-            $container->register($locatorId = $bus.'.messenger.handlers_locator', HandlersLocator::class)
-                ->setArgument(0, $handlersLocatorMappingByBus[$bus] ?? [])
-            ;
+            $locatorId = $bus.'.messenger.handlers_locator';
+            if (!$container->has($locatorId)) {
+                $container->register($locatorId, HandlersLocator::class)
```

Comment:
Could it be interesting to give the handlers locator to the existing definition?
```suggestion
            ($container->hasDefinition($locatorId) ? $container->getDefinition($locatorId) : $container->register($locatorId, HandlersLocator::class))
```

Reply:
Could be but that means the first argument needs to be always be an array and if a new argument is added in Symfony in the future for the `HandlersLocator` then it could be classed as a breaking change because the arguments defined at application level would need to match. 

For example, say a bool is now needed as a second param and someone has already defined the service with a different argument then you'd end up with an error: `Argument #2 ($foo) must be of type string, bool given`

I think it's safer with my suggestion but happy to change it if you prefer.